vault/activity/query.go

```diff
@@ -196,42 +220,45 @@ func (s *PrecomputedQueryStore) Get(ctx context.Context, startTime, endTime time
 	sort.Slice(filteredList, func(i, j int) bool {
 		return filteredList[i].After(filteredList[j])
 	})
-	contiguous := timeutil.GetMostRecentContiguousMonths(filteredList)
-	actualStartTime := contiguous[len(contiguous)-1]
 
-	s.logger.Trace("chose start time", "actualStartTime", actualStartTime, "contiguous", contiguous)
-
-	endTimes, err := s.listEndTimes(ctx, actualStartTime)
-	if err != nil {
-		return nil, err
-	}
-	s.logger.Trace("retrieved end times from storage", "endTimes", endTimes)
-
-	// Might happen if there's a race with GC
-	if len(endTimes) == 0 {
-		s.logger.Warn("missing end times", "start time", actualStartTime)
-		return nil, nil
-	}
-	var actualEndTime time.Time
-	for _, t := range endTimes {
-		if timeutil.InRange(t, startTime, endTime) {
-			if actualEndTime.IsZero() || t.After(actualEndTime) {
-				actualEndTime = t
-			}
+	closestStartTime := time.Time{}
+	closestEndTime := time.Time{}
+	maxTimeDifference := time.Duration(0)
+	for i := len(filteredList) - 1; i >= 0; i-- {
```

Comment:
all nits

1) is there a reason why we're doing a reverse for loop? i think it might be a bit easier to read with a range loop

2) do you think it makes sense to break this off into its own separate function? it might make it a bit easier to test this logic, and it also might make it more understandable for future engineers

i only ask these things because activity log is growing fairly hard to dive into and understand for those less familiar